test/test_jit_cuda_fuser.py

```diff
@@ -2338,6 +2338,32 @@ def t(x: torch.Tensor, y: torch.Tensor, dim: List[int], keepdim: bool):
         self.assertEqual(o, jit_o)
         self.assertGraphContains(t_jit.graph_for(x, y, (0, 1), False), FUSION_GUARD)
 
+    @unittest.skipIf(is_pre_volta(), "reduction not supported in pre volta device")
+    @unittest.skipIf(not RUN_NVFUSER, "requires CUDA")
+    @unittest.skipIf(GRAPH_EXECUTOR != ProfilingMode.PROFILING,
+                     "Requires fusion optimization pass to be effective")
+    def test_profile_ivalue_multiple_profiles(self):
+        dtype = torch.float
+        device = "cuda"
+        x = torch.randn([7, 4, 7], dtype=dtype, device=device)
+
+        def t(x, num: int):
+            for i in range(num):
+                # varying reduction axes should break profile_ivalue
+                tmp = x.sum(i, keepdim=True)
+                # inplace add on input/output, can't be functionalized/fused
+                x += tmp
+            return x
+
+        with nvfuser_singleton_fusion(True):
```

Comment:
Do we need to enable nvfuser_singleton_fusion if function t has multiple operations?

Reply:
We actually do need it here.

There should be a single `sum` node for fusion (assuming that we are not unrolling the loop). So it's safer to enable single node fusion.
The ideas is that we do NOT want the `sum` to be fused, because we see different fusion axes and aren't converting that to a profiled constant.